### PR TITLE
support elastic2

### PR DESCRIPTION
**Pull request by petrjasek**
added there reindex code from elasticsearch, in new version there is an elastic api so it's not there
anymore. it uses new with elastic 2 and the old one with elastic 1.

also testing on travis against both elastic 1 and 2.

### CODE REVIEW

File: eve_elastic/elastic.py [hunk 1]
```diff
@@ -419,15 +428,25 @@ def find(self, resource, req, sub_resource_lookup):
...
+            query['highlight'].setdefault('require_field_match', False)
 
         source_projections = None
         if self.should_project(req):
             source_projections = self.get_projected_fields(req)
 
         args = self._es_args(resource, source_projections=source_projections)
         try:
+            try:
+                print('query', resource, args, json.dumps(query, indent=4))
```
Comment (naspeh) [inline eve_elastic/elastic.py]: Is this `print` needed here?

**Comment by jerome-poisson** [comment]
Hey @petrjasek I'm surprised to see it's working without changing requests as in https://github.com/superdesk/superdesk-core/pull/326 . It's removed only in E.S. 5 ?

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,56 @@
...
+                  packages:
+                      - elasticsearch
+
+        - env: ELASTIC17PYTHON35
+          python: "3.5"
+          addons:
+              apt:
+                  sources:
+                      - elasticsearch-1.7
+                      - mongodb-3.0-precise
```
Comment (naspeh) [inline .travis.yml]: is there `mongodb` needed?

Comment (petrjasek) [inline .travis.yml]: hope not :wink:

File: eve_elastic/elastic.py [hunk 3]
```diff
@@ -428,6 +438,7 @@ def find(self, resource, req, sub_resource_lookup):
         try:
             hits = self.elastic(resource).search(body=query, **args)
         except elasticsearch.exceptions.RequestError as e:
+            raise
```
Comment (mdhaman) [inline eve_elastic/elastic.py]: is `raise` required here?

Comment (petrjasek) [inline eve_elastic/elastic.py]: no, my bad

**Pull request merged**